server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/MinorCompactor.java

```diff
@@ -148,6 +149,13 @@ public CompactionStats call() {
           reportedProblem = true;
         } catch (CompactionCanceledException e) {
           throw new IllegalStateException(e);
+        } catch (Throwable t) {
+          // This is an unexpected situation which is critical. If we cannot minor compact
+          // then this tablet will eventually become unusable if we are still ingesting data.
+          log.error("MinC failed ({}).  Aborting.", t.getMessage(), t);
+          // TODO: Should we simply unload this tablet instead?
+          Halt.halt("Unexpected MinC failure: " + t.getMessage(), 2);
+          throw t;
```

Comment:
We typically try to either log an error or throw, not both.  Instead of halting in MinorCompactor, we may want halt [here in Tablet](https://github.com/apache/accumulo/blob/15a95feebc74bd10e5ba73d4c2ecfc726435bc55/server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java#L829) where we catch Exception or Error.  That code currently does not catch Throwable.

Reply:
Good call.  I will move the halt logic.  The question still remains whether halting is the appropriate action here.  I could try unloading the tablet instead.....

Reply:
Looking closer that might not work because of the IllegalStateException thrown when the minor compaction is cancelled....

Reply:
Following the stack further up, we are already catching Throwable [here in MinorCompactionTask](https://github.com/apache/accumulo/blob/94844b3ef88bc9e02b5cb5c8eeff81411ac8221d/server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/MinorCompactionTask.java#L111).  So if we decide to do anything to the tablet, I think that would be the best place since that method is already operating on the tablet object.  The finally in that try will set the ```minorCompactionState``` so we one option would be to add another state to the [enum in Tablet](https://github.com/apache/accumulo/blob/15a95feebc74bd10e5ba73d4c2ecfc726435bc55/server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java#L217) to indicate failed state.  I don't know what all this state is used for but it may be less complicated than unloading the Tablet and less problematic than halting the tserver.